Tribler/Policies/BoostingSource.py

```diff
@@ -287,8 +282,41 @@ def _on_database_updated(self, dummy_subject, dummy_change_type, dummy_infohash)
         self.database_updated = True
 
     def get_source_text(self):
-        return self.channel.name if self.channel else None
+        return str(self.channel_dict[2]) if self.channel_dict else None
 
+    def _load_torrent(self, infohash, callback=None):
+        """
+        function to load infohash and put in into database.
+        """
+
+        # session is quitting
+        if self.session is None or self.session.lm.torrent_store is None or self.session.get_torrent_store() is None:
+            return
```

Comment:
Yes I have. And that is the case, when I can't find session, and some elements in `LaunchManyCore` is already shutting down